core/math/src/main/java/org/arakhne/afc/math/tree/node/AbstractParentlessTreeNode.java

```diff
@@ -247,7 +248,7 @@ public String toString() {
 	@Override
 	public D[] getAllUserData(D[] array) {
 		if (this.data == null) {
-			return null;
+			return  (D[]) Array.newInstance(array.getClass(),0);
```

Comment:
Yes you are right, we need to check  array first for null. The Generic type D[] confused me as we cannot  do new D[].  I will think of a better solution without involving array parameter.